src/strict/core/signal_engine.py

```
@@ -19,18 +18,79 @@ def __init__(self) -> None:
+        return freq, mag
+
+    @staticmethod
+    def _validate_filter_params(
+        values: list[float] | NDArray,
+        cutoff: float | list[float],
+        fs: float,
+        order: int,
+    ) -> float:
+        """Validate common filter parameters."""
+        if fs <= 0:
+            raise ValueError("Sampling rate (fs) must be positive")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)